### PR TITLE
restore mononoke getdeps integration tests

### PR DESCRIPTION
**Pull request by ahornby**
Bring back mononoke getdeps integration tests.  This is a refesh of the previously working export-D34186407 branch from 2022

Main changes since:
 * depend on restored sapling manifest from previous commit
 * bring back selected manifests needed by mononoke that were deleted in badf6d159dada727acfb5394bd17de223b336975. I added the sqlite binary packages from this to the sqlite3 manifest rather than bring back the sqllite3-bin manifest

Test Plan:

first build dependencies:
```
python3 ./build/fbcode_builder/getdeps.py install-system-deps --recursive mononoke_integration
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests sapling
python3 ./build/fbcode_builder/getdeps.py build --allow-system-packages --no-facebook-internal --src-dir=. --no-tests mononoke
```

then test locally:
```
./build/fbcode_builder/getdeps.py build --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

to iterate on .t:
```
./build/fbcode_builder/getdeps.py build --no-deps --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
./build/fbcode_builder/getdeps.py test --num-jobs 16 --allow-system-packages --no-facebook-internal --src-dir=. mononoke_integration
```

Generate the github actions yaml
```
./build/fbcode_builder/getdeps.py generate-github-actions --allow-system-packages --test --os-type linux --job-file-prefix mononoke-integration_ --job-name-prefix "Mononoke Integration " --output-dir .github/workflows mononoke_integration
```

### CODE REVIEW

File: eden/scm/exec/hgmain/src/chg.rs [hunk 1]
```diff
@@ -129,19 +129,30 @@ fn should_call_chg(args: &[String]) -> (bool, &'static str) {
...
-        }
+    chg_flag_file_decision().unwrap_or((false, "(default fallthrough)"))
+}
 
-        if let Some(etc_decision) = file_decision(Some("/etc/mercurial/usechg")) {
-            return (etc_decision, "/etc/mercurial/usechg");
-        }
-    } else if cfg!(unix) {
-        return (true, "unix");
+#[cfg(all(unix,not(fbcode_build)))]
```
Comment (quark-zju) [inline eden/scm/exec/hgmain/src/chg.rs]: Why this change? It seems the original code is cleaner - no duplication of the function signature.

File: eden/mononoke/tests/integration/library.sh [hunk 2]
```diff
@@ -2225,9 +2225,18 @@ function gitexport() {
 function gitimport() {
   log="$TESTTMP/gitimport.out"
 
+  local git_cmd
+  if type --path git.real > /dev/null; then
```
Comment (quark-zju) [inline eden/mononoke/tests/integration/library.sh:2231]: Just set `NOSCMLOG=1` at some top-level? That applies to subprocesses named `git` or `sl` spawned by `git` or `sl` too.

**Comment by quark-zju** [comment]
Might be easier to review if split into smaller diffs.

File: eden/scm/exec/hgmain/src/chg.rs (hunk 1 above)
Comment (ahornby) [reply eden/scm/exec/hgmain/src/chg.rs]: Original version didn't compile, error due to unknown feature

File: eden/mononoke/tests/integration/library.sh (hunk 2 above)
Comment (ahornby) [reply eden/mononoke/tests/integration/library.sh:2231]: Mononoke defaults to git.real which doesn't exist outside meta. This check detects if we are inside meta (git.real is ok) or outside (use git).

File: eden/scm/exec/hgmain/src/chg.rs (hunk 1 above)
Comment (quark-zju) [reply eden/scm/exec/hgmain/src/chg.rs]: Do you have the error message? It's not clear from the code what feature is unknown (unix? fbcode_build?), or how the new code solves the problem.

Comment (ahornby) [reply eden/scm/exec/hgmain/src/chg.rs]: can't reproduce it now.  may have been due to toolchain on my personal machine.   will try the PRs to facebook/sapling without it

**Comment by ahornby** [comment]
closing as raised PR to https://github.com/facebook/sapling/pull/951
